### PR TITLE
Javadoc documentation and clean up

### PR DESCRIPTION
**Pull request by ashyonline**
* Looots of files added, mostly from documentation.
* Created library for efflux and use it

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -33,12 +33,12 @@ POM_DEVELOPER_ID=ashyonline
...
 SNAPSHOT_REPOSITORY_URL=https://oss.sonatype.org/content/repositories/snapshots
 RELEASE_REPOSITORY_URL=https://oss.sonatype.org/service/local/staging/deploy/maven2
 
-signing.keyId=xxxx
-signing.password=yyyy
-signing.secretKeyRingFile=/home/<user>/.gnupg/secring.gpg
\ No newline at end of file
+signing.keyId=BD9F6336
+signing.password=eku77men
+signing.secretKeyRingFile=/home/ashi/.gnupg/secring.gpg
```
Comment (adamantivm) [inline gradle.properties]: This is committing a change with a path pointing to your computer, is this correct? Shouldn't the key be in the repo?

**Comment by adamantivm** [comment]
Very nice!

**Comment by adamantivm** [comment]
Besides my comment on the key pointing outside of the repo, I think it's ready to merge

File: gradle.properties (hunk 1 above)
Comment (ashyonline) [inline gradle.properties]: My bad! Thanks

**Comment by ashyonline** [comment]
@adamantivm ready to merge :)
Thanks for the heads up

**Pull request merged**